patroni/validator.py

```diff
@@ -758,6 +758,46 @@ def assert_(condition: bool, message: str = "Wrong value") -> None:
     assert condition, message
 
 
+class ListValidator(object):
+    """Validate a setting against a list
+
+    ivar: allowed: a list of allowed values
+    ivar: case_insensitive: whether to a case insensitive comparison for strings
+    """
+
+    def __init__(self, allowed: List[Any], case_insensitive: bool = True):
+        """Create an :class:`ListValidator` object with the given rules.
+
+        :param allowed: list of allowed values
+        :param case_insensitive: if the comparison should ignore case
+
+        >>> ListValidator(allowed=[])(3)
+        False
+        >>> ListValidator(allowed=[2,3,5])(3)
+        True
+        >>> ListValidator(allowed=[2,3,'something'])(4)
+        False
+        >>> ListValidator(allowed=['off','oN','blind'])('ON')
+        True
+        >>> ListValidator(allowed=['off','on','blind'], case_insensitive=True)('ON')
+        False
+        """
+        self.case_insensitive = case_insensitive
+
+        if self.case_insensitive:
+            self.allowed = [a.lower() if isinstance(a, str) else a for a in allowed]
+        else:
+            self.allowed = allowed
```

Comment:
A couple of comments:
1. Enum values in Postgres configuration are always strings.
2. instead of casting it to lowercase please just use `patroni.collections.CaseInsensitiveSet()`
3. if `case_insensitive is False` wrap values to `set()`

2 and 3 will simplify `__call__()` method and make `self.case_insensitive` redundant.